utils/common/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -110,6 +110,22 @@ fun File.unpack(
         }
     }
 
+/**
+ * Try to unpack this [File] of an unknown archive type to [targetDirectory] using [filter] to select only the entries
+ * of interest. Use the file name to guess the archive type. If this is not possible due to a missing or unknown file
+ * extension, try out the supported archive types one after the other.
+ */
+fun File.unpackTryAllTypes(targetDirectory: File, filter: (ArchiveEntry) -> Boolean = { true }) {
+    (listOf(ArchiveType.getType(this.name)) + ArchiveType.values().toList()).distinct().asSequence()
+        .filterNot { it == ArchiveType.NONE }
+        .firstOrNull { archiveType ->
+            val result = runCatching {
+                unpack(targetDirectory, forceArchiveType = archiveType, filter)
+            }
+            result.getOrDefault(false)
```

Comment:
Chain these calls. However, when moving the `runCatching` to `unpack` as proposed in the previous commit you wouldn't need this at all.

Reply:
This is no longer necessary.